source/s3_auto_ranged_get.c

```diff
@@ -520,6 +520,15 @@ static void s_s3_auto_ranged_get_request_finished(
     bool found_object_size = false;
     bool request_failed = error_code != AWS_ERROR_SUCCESS;
 
+    if (meta_request->headers_callback != NULL && request_failed && request->send_data.response_headers != NULL) {
```

Comment:
Will the `discovers_object_size` always be 0 when the `request_failed` is set? Otherwise, it feels like it's possible to invoke the header callback twice.

Reply:
Moved callback invocation to meta request finish handler as suggested in the other comment. Thanks!